s3/src/main/scala/geotrellis/spark/io/s3/S3GeoTiffRDD.scala

```diff
@@ -75,15 +75,16 @@ object S3GeoTiffRDD extends LazyLogging {
     crs: Option[CRS] = None,
     timeTag: String = GEOTIFF_TIME_TAG_DEFAULT,
     timeFormat: String = GEOTIFF_TIME_FORMAT_DEFAULT,
-    maxTileSize: Option[Int] = None,
+    maxTileSize: Option[Int] = Some(256),
```

Comment:
probably we don't need `maxWindowSize`, be careful with fixing tests.

Reply:
Good point on tests. I'm going to remove the configuration, it doesn't seem to serve its purpose now. One thing to call out is that `maxTileSize` is really that. We don't have any way to generate windows of a fixed size, which this class used to do at some point. I would regard that as unspecified behavior though unless there are strong opinions to the contrary.

Reply:
https://github.com/locationtech/geotrellis/blob/master/s3/src/test/scala/geotrellis/spark/io/s3/S3GeoTiffRDDSpec.scala#L176

Max tileSize would be 251 in both rdds in this tests, if you'll pass explicit 1024 it would pass

Reply:
Also remove [this function](https://github.com/locationtech/geotrellis/blob/master/s3/src/test/scala/geotrellis/spark/io/s3/S3GeoTiffRDDSpec.scala#L54) at all :+1: